libs/layout/src/lib/components/accordion/item/accordion-item.component.ts

```diff
@@ -0,0 +1,201 @@
+
+	/**
+	 * Moves the focus to the accordion item above the current one
+	 */
+	@HostListener('keydown.ArrowUp', ['$event']) arrowUp(event: Event) {
+		this.handleWhenFocussed(() => {
+			event.preventDefault();
+			event.stopImmediatePropagation();
+			this.parent.moveFocus(this.id, 'up');
+		});
+	}
+
+	/**
+	 * Moves the focus to the accordion item below the current one
+	 */
+	@HostListener('keydown.ArrowDown', ['$event']) arrowDown(event: Event) {
+		this.handleWhenFocussed(() => {
+			event.preventDefault();
+			event.stopImmediatePropagation();
+			this.parent.moveFocus(this.id, 'down');
+		});
+	}
+
+	/**
+	 * Moves the focus to the first accordion item
+	 */
+	@HostListener('keydown.Home') home() {
+		this.handleWhenFocussed(() => {
+			this.parent.moveFocus(this.id, 'first');
+		});
+	}
+
+	/**
+	 * Moves the focus to the last accordion item
+	 */
+	@HostListener('keydown.End') end() {
+		this.handleWhenFocussed(() => {
+			this.parent.moveFocus(this.id, 'last');
+		});
+	}
+
+	/**
+	 * Whether the accordion item is disabled
+	 */
+	@Input() public disabled: boolean = false;
+
+	/**
+	 * The id of the accordion item
+	 */
+	public readonly id: string = UUID.UUID();
+
+	/**
+	 * Whether the accordion item is open
+	 */
+	public isOpen: boolean = false;
+
+	/**
+	 * Whether the accordion item is focussed
+	 */
+	private hasFocus: boolean = false;
+
+	constructor(
+		private readonly parent: NgxAccordionComponent,
+		private readonly cdRef: ChangeDetectorRef,
+		private readonly renderer: Renderer2
+	) {}
+
+	/**
+	 * Updates the current open/closed state of the accordion item, regardless of the disabled state
+	 */
+	public updateAccordionItemState(isOpen: boolean): void {
+		// Iben: Sets the item to open and updates the parent state
+		this.isOpen = isOpen;
+
+		// Iben: Trigger the visual changes
+		this.cdRef.detectChanges();
+	}
+	/**
+	 * Set the focus on the summary item
+	 */
+	public focus() {
+		this.summaryElement?.nativeElement.focus();
+	}
+
+	/**
+	 * Set the focus state of the accordion item
+	 *
+	 * @param hasFocus - Whether the item has focus
+	 */
+	public setFocus(hasFocus: boolean) {
+		this.hasFocus = hasFocus;
+	}
+
+	/**
+	 * Register the item to its parent
+	 */
+	public ngOnInit(): void {
+		this.parent.registerItem(this);
+	}
+
+	/**
+	 * Listen to the default HTML events of the details object
+	 */
+	public ngAfterViewInit(): void {
+		// Iben: If for some reason no accordion item is found, we return
+		if (!this.detailsElement?.nativeElement) {
+			return;
+		}
+
+		// Iben: Prevent the accordion from being opened if it is disabled
+		this.renderer.listen(this.detailsElement.nativeElement, 'click', (event: Event) => {
+			if (this.disabled) {
+				event.preventDefault();
+				event.stopImmediatePropagation();
+			}
+		});
+
+		// Iben: Listen to the open state of details and update the internal one
+		this.renderer.listen(this.detailsElement.nativeElement, 'toggle', (event: ToggleEvent) => {
```

Comment:
Should these listeners not be removed in the `ngOnDestroy`? And if it isn't required due to the destruction of the `Renderer2` class, shouldn't we explicitly remove the listeners in the same way we usually also push `void` onto the `destroyed$` observable before completion?

Reply:
There is no removeListener method nor does the listen method return a listener to cancel. All I've found in terms of recent information is that once your component gets destroyed (and the instance of Renderer2 along with it), all the listeners get removed as well. In line with how we don't need to remove HostListeners manually either.

It's a good question, I asked myself the same one when implementing it 😂

Reply:
Makes sense! 👍